esss_fix_format/code_formatter.py

```diff
@@ -0,0 +1,421 @@
+'''
+Module to handle code formatting.
+
+There are 3 modes of operation:
+
+1. Always spawning a new java process through the call to:
+
+format_code(code_to_format)

```

Comment:
How will be the day-by-day use? "ff -c" will call the first mode? If so we should update the README

Reply:
I did not add options for users to choose a different mode from the command line (it always uses the third mode). 

It should be reasonably easy to update to the 2nd mode which would start and stop the java process in the same invocation (so, I can go to that mode and if the slowdown is acceptable I can deprecate the third mode, but I don't think the user has to worry about choosing a mode -- either it works well for everyone or it doesn't).

I'm ok in creating a different project for the python formatter and letting `esss_fix_format` just use its public API (as would be the case if we were actually using a different python formatter).